pyprof/nvtx/nvmarker.py

```diff
@@ -350,12 +350,141 @@ def patch_apex():
         import fused_layer_norm_cuda
         patchClass(fused_layer_norm_cuda)
 
+
+# Helper function to dump the passed in dict config as an nvtx
+# marker with "model_config" key
+#
+def push_nvtx_model_config(config):
+    nvtx_msg = json.dumps({"model_config": config})
+    nvtx.range_push(nvtx_msg)
+
+
+# Capture dataloader config (num_workers and pin_memory) and
+# emit a model_config nvtx range with the information
+#
+def patch_dataloader_init():
+    mod = torch.utils.data.dataloader
+    old_init = mod.DataLoader.__init__
+
+    def new_init(self, *args, **kwargs):
+
+        num_workers = kwargs.get("num_workers",0)
+        pin_memory = kwargs.get("pin_memory", False)
+
+        push_nvtx_model_config({"num_workers": num_workers, "pin_memory": pin_memory})
+        old_init(self, *args, **kwargs)
+        nvtx.range_pop()
+
+    mod.DataLoader.__init__ = new_init
+
+
+# Flag to indicate that cudnn_benchmark_disabled has already been reported
+#
+global cudnn_benchmark_disabled_reported
+cudnn_benchmark_disabled_reported = False
+
+# Path the given mod/function so that if it is ever executed and 
+# torch.backends.cudnn.benchmark is not true, it will emit an nvtx
+# range to report that fact
+#
+def patch_with_always_benchmark(mod, fn_name):
```

Comment:
According to this: https://discuss.pytorch.org/t/what-does-torch-backends-cudnn-benchmark-do/5936
we only want to enable this when the input sizes are not varying. Are we accounting for that anywhere before we set this?

Reply:
We aren't accounting for that here. We can handle it in the script that consumes this info.

Reply:
I understand now. Got ahead of myself

Reply:
No, it was an excellent point that I wasn't aware of. Regardless, no change needed here.